jsapp/js/components/common/koboImage.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react'
+import bem, {makeBem} from 'js/bem'
+import LoadingSpinner from 'js/components/common/loadingSpinner'
+import './koboImage.scss'
+
+bem.KoboImage = makeBem(null, 'kobo-image', 'figure')
+bem.KoboImage__image = makeBem(bem.KoboImage, 'image', 'img')
+
+type KoboImageProps = {
+  src: string
+}
+
+type KoboImageState = {
+  isLoading: boolean
+}
+
+/**
+ * An image component that handles loading the image with a spinner.
+ */
+class KoboImage extends React.Component<KoboImageProps, KoboImageState> {
+  constructor(props: KoboImageProps){
+    super(props)
+    this.state = {
+      isLoading: true
+    }
+  }
+
+  componentDidMount() {
+    this.preloadImage()
+  }
+
+  componentWillReceiveProps() {
+    this.preloadImage()
+  }
+
+  preloadImage() {
+    this.setState({isLoading: true})
```

Comment:
same as above

Reply:
In audio you're right, but here we can receive a `preloadImage` request via `componentWillReceiveProps`. This could happen if for some reason the image will change after `KoboImage` is rendered. I don't think we have such a case now in code, but it could happen in future, so let's leave it :)